### DISCUSSION TITLE
Different releases of Tasmota, how to mix?

### DISCUSSION THREAD

**Discussion by whabby** [arendst/Tasmota#20343] [Q&A] [answered]
Hi everybody

To control my water reservoir, I would like to integrate the following modules into Tasmota and flash them onto an ESP32-CAM. I would like to use this module to be able to use the camera for optical control of the tank.

I would also like to install the following components:
- 2 relays for pump control
- WS 2812 Led-Stripe for optical level indication
- SR04 module for level measurement

To be able to use the camera, I would like to use Tasmota "Webcam". Is there a way to integrate the WS2812 and the SR04 in this release? These would be available in the release "Sensors".

Thanks for your help!

**Comment by jkwim** [comment]
If you do your own compiling, theoretically you should be able to combine specific features. I have used different combinations on ESP8266.

The main limitation is to manage the overall binary size and the RAM usage.

Haven't done anything on top of ESP32 Webcam myself so I cannot comment there.

You would need a Linux environment to build/compile. A virtual machine or a Single Board Computer (SBC) like Raspberry Pi, Orange Pi to setup the compiling environment.

[https://tasmota.github.io/docs/Create-your-own-Firmware-Build-without-IDE](https://tasmota.github.io/docs/Create-your-own-Firmware-Build-without-IDE/)

**Reply by barbudor** [reply, marked as answer]
@whabby 

You can also use TasmoCompiler, an online tool which allows to create your own binary configuration

Alternatively, use GitPod, a online development tool which avoid you to install anything on your computer bu you need some basic skills in order to use such tool.

And finally if you want to do that on your own computer, install VSCode environment. The doc includes a link to setup VSCode without installation in "Portable mode"

You DON'T need a linux environment. VSCode and Platform IO work perfectly fine also on Windows and Mac
A standard PC/Mac will be much more powerfull than a RaspberryPi to compile a large software like Tasmota

**Comment by Wintespe** [comment]
**Enable Portable mode for  VS Code**

After unzipping the VS Code download, simply create afolder named **Data**  within code's folder

From then on, that folder will be used to contain all code data, including session state, preferences, extensions, etc.

The data folder can be moved to other VS Code installations. This is useful for updating your portable Code version: simply move the data folder to a newer extracted version of VS Code.

**Comment by whabby** [comment]
Hi everybody
Problem solved using Gitpod. Thanks very much, I learned a lot again !

Regards whabby

 

Von: Barbudor ***@***.***> 
Gesendet: Sonntag, 7. Januar 2024 18:42
An: arendst/Tasmota ***@***.***>
Cc: whabby ***@***.***>; Mention ***@***.***>
Betreff: Re: [arendst/Tasmota] Different releases of Tasmota, how to mix? (Discussion #20343)

 

@whabby <https://github.com/whabby> 

You can also use TasmoCompiler, an online tool which allows to create your own binary configuration

Alternatively, use GitPod, a online development tool which avoid you to install anything on your computer bu you need some basic skills in order to use such tool.

And finally if you want to do that on your own computer, install VSCode environment. The doc includes a link to setup VSCode without installation in "Portable mode"

You DON'T need a linux environment. VSCode and Platform IO work perfectly fine also on Windows and Mac
A standard PC/Mac will be much more powerfull than a RaspberryPi to compile a large software like Tasmota
